sstables/sstables.hh

```diff
@@ -902,6 +902,18 @@ public:
         }
         const stats_metadata& s = *static_cast<stats_metadata *>(p.get());
         return s;
+    }
+     stats_metadata& get_stats_metadata_for_write() {
```

Comment:
Doing sstable statistics modifications like this:
```c++
auto& stats = sst->get_stats_metadata_for_write();
stats.some_field = new_value;
sst->rewrite_statistics();
```
Is error prone.

It would be better to have something like this:
```
future<> mutate_statistics(std::function<void(stats_metadata& mutable_stats)> mutate_func) {
    // pass mutable statistics to method, then save it on disk
}
```

Reply:
sounds good.

Reply:
Done. I ended up with a helper to avoid in place modification.